From a6bea05508f850d70f87ae38b6f3b1a5bb8aa489 Mon Sep 17 00:00:00 2001
From: Fabien Chereau <fabien.chereau@intel.com>
Date: Wed, 11 Feb 2015 11:46:34 +0100
Subject: [PATCH] toolchain: fix buggy shell behaviour on unbutu after sourcing
 environment-setup

This fixes an issue for end users using the toolchain on ubuntu, when they
type an invalid command in the shell.

The environment setup script generated with the cross compilation toolchain
redefines the PYTHONHOME environment variable to use the local python version.

On ubuntu, typing an unknown command in a shell will run the
ubuntu-specific command_not_found_handle python script. The problem is that
this script is not compatible with the python version shipped with the cross
compilation toolchain.

Signed-off-by: Fabien Chereau <fabien.chereau@intel.com>
---
 meta/classes/toolchain-scripts.bbclass | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/meta/classes/toolchain-scripts.bbclass b/meta/classes/toolchain-scripts.bbclass
index b9f2aea9..886886e 100644
--- a/meta/classes/toolchain-scripts.bbclass
+++ b/meta/classes/toolchain-scripts.bbclass
@@ -68,6 +68,8 @@ toolchain_shared_env_script () {
 	echo 'export OECORE_SDK_VERSION="${SDK_VERSION}"' >> $script
 	echo 'export ARCH=${ARCH}' >> $script
 	echo 'export CROSS_COMPILE=${TARGET_PREFIX}' >> $script
+	echo '' >> $script
+	echo 'unset command_not_found_handle' >> $script
 }
 
 #we get the cached site config in the runtime
-- 
2.1.0

